src/adapter.rs

```diff
@@ -1163,7 +1163,21 @@ impl<'a> Adapter<'a> for RustdocAdapter<'a> {
                                             token.as_impl().expect("not an Impl token");
 
                                         if let Some(path) = &impl_token.trait_ {
-                                            if let Some(item) = item_index.get(&path.id) {
+                                            // When the implemented trait is from the same crate
+                                            // as its definition, the trait should be present
+                                            // in `item_index`. Otherwise, the
+                                            // `rustdoc_types::Trait` is not in this rustdoc. 
+                                            // The needed foreign traits for lints are hand-written 
+                                            // in `dummy_trait_items`.
+                                            // See the comments in `src/indexed_crate.rs`
+                                            // for details.
+                                            let found_item = item_index
+                                                .get(&path.id)
+                                                .or(current_crate.dummy_trait_items.get(&path.id))
+                                                .or(previous_crate.and_then(|crate_| {
+                                                    crate_.dummy_trait_items.get(&path.id)
+                                                }));
+                                            if let Some(item) = found_item {
                                                 Box::new(std::iter::once(
                                                     origin.make_implemented_trait_token(path, item),
                                                 ))
```

Comment:
Should the `trait: Trait!` in `ImplementedTrait` in the schema be replaced with `trait: Trait` now that there is no guarantee that a `Trait` exists?

Reply:
Yes, good catch.

Reply:
Done in c25fe8efd190586972cbadb430e8840764be0de7.